### PR TITLE
Eliminate Sequence.SubSequence + Use @_borrowed

### PR DESCRIPTION
**Pull request by airspeedswift**
These affected very similar benchmarks so trying them together.

### CODE REVIEW

**Comment by airspeedswift** [comment]
@swift-ci please test

**Comment by airspeedswift** [comment]
@swift-ci please benchmark

File: stdlib/public/core/Collection.swift [hunk 1]
```diff
@@ -332,7 +332,7 @@ extension IndexingIterator: IteratorProtocol, Sequence {
 /// or bidirectional collection must traverse the entire collection to count
 /// the number of contained elements, accessing its `count` property is an
 /// O(*n*) operation.
-public protocol Collection: Sequence where SubSequence: Collection {
+public protocol Collection: Sequence {
   // FIXME: ideally this would be in MigrationSupport.swift, but it needs
   // to be on the protocol instead of as an extension
   @available(*, deprecated/*, obsoleted: 5.0*/, message: "all index distances are now of type Int")
```
Comment (slavapestov) [inline stdlib/public/core/Collection.swift]: Is this still the case?

**Comment by rjmccall** [comment]
Okay, there are some perf numbers in there if you look at the log, although they don't seem particularly credible.

File: stdlib/public/core/Collection.swift (hunk 1 above)
Comment (airspeedswift) [reply stdlib/public/core/Collection.swift]: You mean the commented out obsolete part? I need to go through and remove all those – but that's not an ABI thing so not as urgent as some other stuff.

**Comment by airspeedswift** [comment]
@DougGregor getting this really strange error on the build bot that doesn't reproduce on any other machines:

```
22:34:43 [ RUN      ] FlattenCollection.FlattenCollection instances (BidirectionalCollection)
22:34:43 stderr>>> Assertion failed: (registered == c && "objc_readClassPair failed to instantiate the class in-place"), function swift_instantiateObjCClass, file /Users/buildnode/jenkins/workspace/swift-PR-osx/branch-master/swift/stdlib/public/runtime/SwiftObject.mm, line 1253.
22:34:43 stderr>>> CRASHED: SIGABRT
22:34:43 the test crashed unexpectedly
22:34:43 [     FAIL ] FlattenCollection.FlattenCollection instances (BidirectionalCollection)
```

Any ideas what might cause that?

**Comment by DougGregor** [comment]
@airspeedswift that can happen if two Swift classes have the same @objc names. Prior to macOS Mojave, `objc_readClassPair` would return a null if the class was already registered, causing Swift's standard library to assert.

**Pull request closed without merging**